script.R

```diff
@@ -10,6 +10,53 @@ if_error <- function(x, y = NULL) {
 `%||%` <- function(x, y) {
   if (!length(x) || is.null(x)) y else x
 }
+check_if_pkg_available <- function(pkg, ver = NULL) {
+  if (is.null(ver)) {
+    length(available.packages(filter = list(add = TRUE, function(db) db[db[, "Package"] == pkg, ]))) > 0
+  } else {
+    length(
+      available.packages(
+        filter = list(
+          add = TRUE,
+          function(db) {
+            db[db[, "Package"] == pkg & db[, "Version"] == ver, ]
+          }
+        )
+      )
+    ) > 0
+  }
+}
+add_to_minicran <- function(x, minicran_path) {
+  temp_dir <- tempfile()
+  on.exit(unlink(temp_dir))
+  dir.create(temp_dir)
+  new_file_name <- gsub("(.*?_.*)_.*?(\\..*)", "\\1\\2", basename(x))
+  file.copy(x, file.path(temp_dir, new_file_name))
+  miniCRAN::addLocalPackage(gsub("_.*", "", basename(new_file_name)), temp_dir, minicran_path)
```

Comment:
Isn't `new_file_name` already a `basename` of `x` so there should be no path to be removed from `new_file_name`.
Also, didn't the previous `gsub()` https://github.com/insightsengineering/r-revdepcheck-action/pull/4/files#r1784222144 remove all `_` signs? What is the point of another removal?
```suggestion
  miniCRAN::addLocalPackage(gsub("_.*", "", new_file_name), temp_dir, minicran_path)
```

Reply:
nope. the previous gsub only removes the third one (and everything what comes after). It also keeps file extension. 
Here, we need to have a package name. I take it from the file name. Alternatively I can unpack the `.tar.gz` file and read DESCRIPTION but this is much complex. I think I'm going to leave it as is.

Reply:
alrighty, thanks for the explanation